jerry-core/ecma/base/ecma-helpers-value.c

```diff
@@ -540,6 +558,21 @@ ecma_make_string_value (const ecma_string_t *ecma_string_p) /**< string to refer
   return ecma_pointer_to_ecma_value (ecma_string_p) | ECMA_TYPE_STRING;
 } /* ecma_make_string_value */
 
+#ifndef CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN
+/**
+ * Symbol value constructor
+ *
+ * @return ecma-value representation of the string argument
+ */
+inline ecma_value_t JERRY_ATTR_PURE JERRY_ATTR_ALWAYS_INLINE
+ecma_make_symbol_value (const ecma_string_t *ecma_symbol_p) /**< symbol to reference in value */
+{
+  JERRY_ASSERT (ecma_symbol_p != NULL);
```

Comment:
Please add an assert that string type is symbol. Also add a check for ecma_make_string_value to exclude symbols.

Reply:
The first part is OK for me, but the second is not. Since all symbol is a special type `ecma-string` and the whole property management system depends on that the prop-name is `ecma-string` and these strings are passed via `ecma_values` this assert prevents to use symbols as property names.